packages/shared/src/hooks/integrations/useIntegration.ts

```diff
@@ -0,0 +1,111 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  REMOVE_INTEGRATION_MUTATION,
+  REMOVE_SOURCE_INTEGRATION_MUTATION,
+  UserIntegration,
+  UserSourceIntegration,
+} from '../../graphql/integrations';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { usePrompt } from '../usePrompt';
+import {
+  deleteIntegrationPromptOptions,
+  deleteSourceIntegrationPromptOptions,
+} from '../../lib/integrations';
+import { gqlClient } from '../../graphql/common';
+
+export type UseIntegration = {
+  removeIntegration: ({
+    integrationId,
+  }: {
+    integrationId: string;
+  }) => Promise<void>;
+  removeSourceIntegration: ({
+    sourceId,
+    integrationId,
+  }: {
+    sourceId: string;
+    integrationId: string;
+  }) => Promise<void>;
+};
+
+export const useIntegration = (): UseIntegration => {
```

Comment:
Why did you split between singular and not?

Reply:
yeah, naming wise maybe a bit confusing but basically useIntegrations is query to list integration and this hook is more of a util for mutations, I might refactor query ones to `useIntegrationsQuery` and so on.

Reply:
Renamed all query based hooks to end with `Query` so they don't overlap with utility hooks 438fa52b6d4f30cb652b2ea5ba8b3ae2c1677c9f